service/storage_proxy.cc

```diff
@@ -3974,18 +3974,62 @@ void storage_proxy::handle_read_error(std::exception_ptr eptr, bool range) {
 }
 
 future<storage_proxy::coordinator_query_result>
-storage_proxy::query_singular(lw_shared_ptr<query::read_command> cmd,
+storage_proxy::query_singular_nonvector(lw_shared_ptr<query::read_command> cmd,
+        dht::partition_range&& pr,
+        db::consistency_level cl,
+        storage_proxy::coordinator_query_options query_options,
+        schema_ptr&& schema,
+        clock_type::time_point timeout,
+        db::read_repair_decision repair_decision) {
+    bool is_read_non_local = false;
+
+    const auto tmptr = get_token_metadata_ptr();
+
+    auto token_range = dht::token_range::make_singular(pr.start()->value().token());
+    auto it = query_options.preferred_replicas.find(token_range);
+    const auto replicas = it == query_options.preferred_replicas.end()
+        ? inet_address_vector_replica_set{} : replica_ids_to_endpoints(*tmptr, it->second);
+
+    auto exec = get_read_executor(cmd, schema, std::move(pr), cl, repair_decision,
+                                            query_options.trace_state, replicas, is_read_non_local,
+                                            query_options.permit);
+
+    if (is_read_non_local) {
+        get_stats().reads_coordinator_outside_replica_set++;
+    }
+
+    return exec->execute(timeout).then_wrapped([p = shared_from_this(), rex = exec, pr = std::move(pr), token_range = std::move(token_range), repair_decision, tmptr] (
```

Comment:
pr = std::move(pr) is use-after-move, but a benign one given that moved pr isn't used within the continuation. better to remove it to prevent someone from accidentally using it in the future

Reply:
other than that minor concern, patch looks good to me.